torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -566,7 +574,11 @@ void ProcessGroupNCCL::workCleanupLoop() {
          /* no increment*/) {
       auto& work = *it;
       if (work->isCompleted()) {
-        // Remove all Completed WorkNCCL Objects from the Vector
+        // Handle Exceptions on failed GPU operations and remove completed
+        // workNCCL objects from work vector.
+        if (work->finishedGPUExecution()) {
```

Comment:
Why do we need to condition this on `work->finishedGPUExecution`?

Reply:
Removing this in latest version